kmod/patch/Makefile

```diff
@@ -1,5 +1,5 @@
 KPATCH_BUILD ?= /lib/modules/$(shell uname -r)/build
-KPATCH_MAKE = $(MAKE) -C $(KPATCH_BUILD) M=$(PWD)
+KPATCH_MAKE = $(MAKE) -C $(KPATCH_BUILD) M=$(PWD) CFLAGS_MODULE='$(KBUILD_CFLAGS_MODULE)'
```

Comment:
@kamalesh-babulal : in my mind, kernel kbuild is a giant ball of spaghetti.  Do we know if the `-mcmodel=large` flag below ever worked?  Or maybe recent churn in kbuild stopped paying attention to the `KBUILD_CFLAGS_MODULE` environment variable?

Reply:
@joe-lawrence I did a quick test, of building a livepatching module using `meminfo.patch` with and without the `KBUILD_CFLAGS_MODULE` being passed via this Makefile. The only difference I see that the `-mcmodel=large` flag is being passed twice with `KBUILD_CFLAGS_MODULE` set. `gcc -Wp,-MMD,/root/.kpatch/tmp/patch/.livepatch-meminfo.mod.o.d ............ -mcmodel=medium .... -I/root/kpatch/kmod/patch -mcmodel=large -fplugin=/root/kpatch/kpatch-build/gcc-plugins/ppc64le-plugin.so ... -DMODULE -mno-save-toc-indirect -mcmodel=large -mcmodel=large -DKBUILD_BASENAME='"livepatch_meminfo.mod"' -DKBUILD_MODNAME='"livepatch_meminfo"' -D__KBUILD_MODNAME=kmod_livepatch_meminfo -c -o /root/.kpatch/tmp/patch/livepatch-meminfo.mod.o /root/.kpatch/tmp/patch/livepatch-meminfo.mod.c`.


I loaded the module built without the `KBUILD_CFLAGS_MODULE += -mcmodel=large` flag and seems to okay. I guess, we can remove the arch specific flag from the Makefile.

Reply:
@kamalesh-babulal : thanks for running those tests.  Were they on top of this PR or against the master branch?
Then if `KBUILD_CFLAGS_MODULE += -mcmodel=large` is not needed in this Makefile, I can tack on another commit to remote it.

Reply:
@joe-lawrence It was run against `master`. I re-ran it against `master + this PR`,  verified the same two cases. 
- First, where the `KBUILD_CFLAGS_MODULE += -mcmodel=large` flag is passed in the Makefile and the module building cmdline appended with `CFLAGS_MODULE='-mcmodel=large'`. It also adds one more `-mcmodel=large`, to Kernel Build cmdline, making it three consecutive times.

- Second, removing the `KBUILD_CFLAGS_MODULE += -mcmodel=large` flag in the Makefile, the cmdline used to build the module, is passed with an empty `CFLAGS_MODULE=''` flag and `-mcmodel=large` is just passed one.

> Then if KBUILD_CFLAGS_MODULE += -mcmodel=large is not needed in this Makefile, I can tack on another commit to remote it.

From the simple `meminfo.patch` test, it looks safe to remove it.

Reply:
I agree, though I am really confused right now as I was having problems passing `-DKLP_REPLACE_ENABLE=false` on 5.14 as it is currently coded.  (I thought I accounted for the inadvertent leading space, but I can re-check).  Kamalesh's test results confirm that it should have worked... so let me re-test commit again.

Reply:
Hello, I had also concluded that the '-R' (non-replace) didn't really work, and I was planning on spending some time today to figure out what might be wrong. Fortunately, I took a look at the pull requests, and look; you had already solved it. Thanks! I understand from the above that there is some discussions as to where the CFLAGS_MODULE patch should go, but I can confirm that the current original patch does the job.
FYI, I tested on a Ubuntu 20.04 with the kmod/patch/Makefile patch applied, with three atomic patches, and it worked as it should; I could install and remove, atomically, the three patches.